Makefile

```diff
@@ -110,3 +115,32 @@ docker-run-node: ## Runs the Miden node as a Docker container
                -v miden-db:/db \
                -v "$${ABSOLUTE_PATH}:/miden-node.toml" \
                -d miden-node-image
+
+# --- remote prover -------------------------------------------------------------------------------
+
+.PHONY: clean-prover
+clean-prover: ## Uninstall prover
+	cargo uninstall miden-proving-service || echo 'prover not installed'
+
+.PHONY: prover
+prover: setup-miden-base update-prover-branch build-prover ## Setup prover directory
+
+.PHONY: setup-miden-base
+setup-miden-base: ## Clone the miden-base repository if it doesn't exist
+	if [ ! -d $(PROVER_DIR) ]; then git clone $(PROVER_REPO) $(PROVER_DIR); fi
+
+.PHONY: update-prover-branch
+update-prover-branch: setup-miden-base ## Checkout and update the specified branch in miden-base
+	cd $(PROVER_DIR) && git checkout $(PROVER_BRANCH) && git pull origin $(PROVER_BRANCH)
+
+.PHONY: build-prover
+build-prover: update-prover-branch ## Update dependencies and build the prover binary with specified features
+	cd $(PROVER_DIR) && cargo update && cargo build --bin miden-proving-service --locked $(PROVER_FEATURES_TESTING) --release
+
+.PHONY: start-prover
+start-prover: ## Run prover. This requires the base repo to be present at `miden-base`
+	cd $(PROVER_DIR) && RUST_LOG=info cargo run --bin miden-proving-service $(PROVER_FEATURES_TESTING) --release --locked -- start-worker -p $(PROVER_PORT) --batch-prover
+
+.PHONY: kill-prover
+kill-prover: ## Kill prover process
+	pkill miden-tx-prover || echo 'process not running'
```

Comment:
I'll be honest I'm not a big fan of this. I don't really want to do spooky things with other repos at random locations.

Once again, I would much prefer this service be in-process for convenience.

Reply:
I don't like this neither. I would even prefer to use a deployed prover somewhere instead, though it would stall CI test jobs if it is down, so I don't think that this is a great idea.

Having the worker as a library and basically starting up as a component here looks like the best shot